src/modules/jsUtils.js

```diff
@@ -0,0 +1,28 @@
+function compareObjects(o1, o2) {
+    // If either are undefined, return false - don't consider undefined to equal undefined.
+    if (typeof o1 === 'undefined' || typeof o1 === 'undefined') {
+        return false;
+    }
+
+    if (Array.isArray(o1)) {
+        if (!Array.isArray(o2)) {
+            return false;
+        }
+        // Simple array comparison - expects same order and only scalar values
+        return o1.length === o2.length && o1.every(function (v, i) {
```

Comment:
I greatly improved the object comparison method. It now supports nested objects and supports comparing order or not.